libs/shared/advert-card/src/lib/shared-advert-card.tsx

```diff
@@ -73,63 +91,211 @@ export const SharedAdvertCard = (props: SharedAdvertCardProps) => {
+          <button
+            type="button"
+            onClick={() => props.onViewClick(props.author._id)}
+          >
+            <img
+              className="card-img"
+              src={props.author.img}
+              alt=""
+              height="40"
+              width="40"
+            />
+          </button>
 
-        <div className="card-title">
-          <a href="url" onClick={() => props.onViewClick(props.author._id)}>
+          <div className="card-title">
+            <a href="url" onClick={() => props.onViewClick(props.author._id)}>
+              <span>
+                {`
+                    ${props.author.firstName || ''} ${
+                  props.author.lastName || ''
+                }
+                  `}
+              </span>
+            </a>
             <span>
-              {`
-								${props.author.firstName || ''} ${props.author.lastName || ''}
-							`}
+              posted on{' '}
+              <TextField
+                inputRef={register({})}
+                id="title"
+                name="title"
+                defaultValue={props.title}
+              />
             </span>
-          </a>
 
-          <span>posted on {props.title}</span>
-          <p>{`${timeCalculator(props.createAt)}`}</p>
+            <p>{`${timeCalculator(props.createAt)}`}</p>
+          </div>
+          <div className="editing-holder editing">
+            <button type="submit">
+              <SaveIcon />
+            </button>
+          </div>
         </div>
-      </div>
 
-      <div className="card-content">
-        <p>{props.description}</p>
-        <img src={props.imgUrl} alt="" />
-      </div>
+        <div className="card-content">
+          <TextField
+            inputRef={register({})}
+            name="description"
+            id="description"
+            multiline
+            fullWidth
+            defaultValue={props.description}
+          />
+          <div className="image-container draggable-container">
+            <img
+              className="files-preview-container__image"
+              src={
+                (!!uploadedImg && uploadedImg.toString()) ||
+                uploadedImg.toString()
+              }
+              alt=""
+            />
+            <div className="middle">
+              <Typography
+                variant="body1"
+                component="p"
+                align="center"
+                gutterBottom
+              >
+                Drag and Drop Images Here
+              </Typography>
+              <SystemUpdateAltIcon display="inline" />
+              <TextField
+                className="file-browser-input"
+                inputRef={register({})}
+                type="file"
+                id="file-browser-input"
+                name="advertImage"
+                onDragOver={(e) => {
+                  e.preventDefault();
+                  e.stopPropagation();
+                }}
+                onDrop={onFileLoad.bind(this)}
```

Comment:
it not necessary bind, if you using arrow function onDrop={onFileLoad}

same and next event